s3-access-grants-plugin/src/main/java/com/amazonaws/s3accessgrants/plugin/internal/S3AccessGrantsStaticOperationDetails.java

```diff
@@ -197,6 +198,14 @@ else if (request instanceof CopyObjectRequest) {
                 s3Prefix = "s3://" + copyObjectRequest.getSourceBucketName() + getCommonPrefixFromMultiplePrefixes(keysList);
             }
         }
+        else if (request instanceof ListPartsRequest) {
+            ListPartsRequest listPartsRequest = (ListPartsRequest) request;
+            if (listPartsRequest.getKey() == null){
+                s3Prefix = "s3://" + listPartsRequest.getBucketName();
+            } else {
+                s3Prefix = "s3://" + listPartsRequest.getBucketName() + "/" + listPartsRequest.getKey();
```

Comment:
Do we already have a code guard to verify if bucketName is not null? I mean if the listPartsRequest by default does not accept a null bucket name, we should be good. I just want to be sure because this is v1 and it would be better if we double check if SDK rejects such requests by default.

Reply:
SDK does not allow you to create such requests. It checks for the necessary parameters before making the API call.